### ISSUE TITLE
Debug branch: Assertion violation at ../src/math/lp/lar_core_solver.h Line: 760

### ISSUE THREAD

**Issue by muchang** [Z3Prover/z3#3311]
Hi,
For this formula:
```
(declare-fun a ()  Real)
(declare-fun b () Real) 
(declare-fun c () Real)  
(declare-fun d () Real) 
(declare-fun e () Real) 
(declare-fun f () Real) 
(declare-fun i () Real)  
(declare-fun g () Real) 
(assert (not (exists (( h Real )) 
  (=> (and (xor (and (= (-  (div 1 h)) 0.0) (= a 2.0 )) 
  ( = e 2.0 )) (= i f g)) 
  (distinct 0.0 b) (=> (< h b) (<= 0.0 (mod (* c h) d ))) (<= 0 f)))))
(check-sat)
```
Z3 debug branch  throws out an assertion violation:
```
ASSERTION VIOLATION
File: ../src/math/lp/lar_core_solver.h
Line: 760
l <= u
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
```

OS: Ubuntu 18.04
Commit: 3ac9dbd

**Comment by levnach** [comment]
Fixed

**Comment by zhendongsu** [comment]
It seems to still crash on the latest debug branch commit: 

```
[592] % z3debug bug.smt2
ASSERTION VIOLATION
File: ../src/math/lp/lar_core_solver.h
Line: 760
l <= u
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
a
[593] % 
```

**Comment by levnach** [comment]
Fixed in a1c36084b8a7847335d212839ddab7fdde62f441
